### ISSUE TITLE
Support docs & components from source directory & dependencies

### ISSUE THREAD

**Issue by axe312ger** [doczjs/docz#1317]
## Situation

I am working on a [set of components](https://github.com/axe312ger/gatsby-mdx-suite) which can be easily reused over several Gatsby projects. Additionally I create project specific components.

To provide a proper documentation to the team of other devs & editors, I want to display documentation from my project specific components plus the ones I included from the (community) repositories.

## Solution

With these 3 pull requests, my use case would be fully covered:

- [x] #1314 
- [x] #1315 
- [x] #1316 

## Todo

The naming of the `customPattern` should probably be changed to a more clear name 🤔 

The code for sure is not bullet proof yet, it also needs some documentation. I also found several other configuration options within docz-core which are not yet reflected on the website. We could tackle these as well :)

Would love to see this integrated, I am very open to feedback to my PRs.

👋 

### Example

With all PRs applied and the config below I can support components and docs from normal projects and mono/lerna/yarn workspace repositories as well.

```js
{
  resolve: `gatsby-theme-docz`,
  options: {
    files: [
      `src/components/mdx/**/docs/*.{md,markdown,mdx}`,
      `node_modules/@gatsby-mdx-suite/*/docs/*.{md,markdown,mdx}`,
      `../../node_modules/@gatsby-mdx-suite/*/docs/*.{md,markdown,mdx}`,
    ],
    customPattern: [
      `src/components/mdx/**/*.{js,jsx,mjs}`,
      `node_modules/@gatsby-mdx-suite/**/*.{js,jsx,mjs}`,
      `../../node_modules/@gatsby-mdx-suite/**/*.{js,jsx,mjs}`,
    ],
  },
}
```

**Comment by rakannimer** [comment]
Thank you for the PRs, I'm sure many people will find them very useful 👍 

I merged #1315 and added minor reviews for #1314 and #1316.

Regarding the docs, if you agree with me and we go with `docgenConfig.searchPatterns` than we could add docs for the complete docgenConfig object in docz.site (https://github.com/doczjs/docz-website) 

I opened an issue there to track the progress : https://github.com/doczjs/docz-website/issues/90

**Comment by axe312ger** [comment]
working now for me in alpha, will report back if issues arise. Thx for the alpha release! :)
